examples/base_api.py

```diff
@@ -0,0 +1,86 @@
+"""A run through of base API operations on all supported clouds."""
+
+import os
+from contextlib import suppress
+
+import pycloudlib
+from pycloudlib.cloud import BaseCloud
+
+
+cloud_config = """#cloud-config
+runcmd:
+  - echo 'hello' > /home/ubuntu/example.txt
+"""
+
+
+def exercise_api(client: BaseCloud):
+    """Run through supported functions in the base API."""
+    try:
+        image_id = client.released_image('focal')
+    except NotImplementedError:
+        image_id = client.daily_image('focal')
+    print('focal image id: {}'.format(image_id))
+    print('launching instance...')
+    instance = client.launch(
+        image_id=image_id,
+        user_data=cloud_config
+    )
+    assert repr(instance) == repr(client.get_instance(instance.name))
+
+    print('instance name: {}'.format(instance.name))
+    with suppress(NotImplementedError):
+        print('instance ip: {}'.format(instance.ip))
+
+    print('starting instance...')
+    instance.start()
+    print('waiting for cloud-init...')
+    instance.execute('cloud-init status --wait --long')
```

Comment:
Maybe we could call `instance.wait()`  here directly, since most instance types do implement the `cloud-init` check in the wait method